sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java

```diff
@@ -0,0 +1,512 @@
+        return coalescedAndSortedResults.map(results -> {
+            List<List<ScoreTuple>> componentScoresInternal = new ArrayList<>();
+            for (int i = 0; i < results.get(0).getComponentScores().size(); i++) {
+                componentScoresInternal.add(new ArrayList<>());
+            }
+            List<Double> undefinedComponentScores = new ArrayList<>();
+            for (int i = 0; i < componentScoresInternal.size(); i++) {
+                undefinedComponentScores.add(-999999.0); // Default undefined value
+            }
+            for (int i = 0; i < results.size(); i++) {
+                List<Double> componentScores = results.get(i).getComponentScores();
+                // TODO: # Remove this small fix after the backend changes are released to deal with empty component score scenarios
+                if (componentScores.isEmpty()) {
+                    componentScores = undefinedComponentScores;
+                }
+                for (int j = 0; j < componentScores.size(); j++) {
+                    ScoreTuple scoreTuple = new ScoreTuple(componentScores.get(j), i);
+                    componentScoresInternal.get(j).add(scoreTuple);
+                }
+            }
+            //Sort scores in descending order
+            for (List<ScoreTuple> scoreTuples : componentScoresInternal) {
+                scoreTuples.sort(Comparator.comparing(ScoreTuple::getScore, Comparator.reverseOrder()));
+            }
+            return componentScoresInternal;
+        });
+    }
+
+    private Flux<Document> getComponentQueryResults(List<FeedRangeEpkImpl> targetFeedRanges, int initialPageSize, DocumentCollection collection, Flux<QueryInfo> rewrittenQueryInfos) {
+        return rewrittenQueryInfos.flatMap(queryInfo -> {
+            Map<FeedRangeEpkImpl, String> partitionKeyRangeToContinuationToken = new HashMap<>();
+            for (FeedRangeEpkImpl feedRangeEpk : targetFeedRanges) {
+                partitionKeyRangeToContinuationToken.put(feedRangeEpk,
+                    null);
+            }
+            documentProducers = new ArrayList<>();
+            super.initialize(collection,
+                partitionKeyRangeToContinuationToken,
+                initialPageSize,
+                new SqlQuerySpec(queryInfo.getRewrittenQuery()));
+
+            return Flux.fromIterable(documentProducers)
+                .flatMap(DocumentProducer::produceAsync)
+                .flatMap(response -> Flux.fromIterable(response.pageResult.getResults()));
+        });
+    }
+
+    private Flux<QueryInfo> retrieveRewrittenQueryInfos(List<QueryInfo> componentQueryInfos) {
+        return aggregatedGlobalStatistics.hasElement().flatMapMany(globalStatistics -> {
+            if (globalStatistics != null) {
+                List<Mono<QueryInfo>> rewrittenQueryInfosInternal = new ArrayList<>();
+                for (QueryInfo queryInfo : componentQueryInfos) {
+                    assert queryInfo.hasOrderBy();
```

Comment:
I thought for vector search component query, there will be be no orderby as it is defined in the policy?

Reply:
That was a previous bug which is now fixed in the backend and deployed.